ginisdk/src/main/java/net/gini/android/authorization/PubKeyManager.java

```diff
@@ -70,17 +95,41 @@ private String getEncodedPublicKey(X509Certificate certificate) {
         return new BigInteger(1, publicKey.getEncoded()).toString(16);
     }
 
-    private Boolean isValidCertificate(X509Certificate remoteCertificate) {
-        String remoteEncoded = getEncodedPublicKey(remoteCertificate);
-        Boolean isValid = false;
+    private String getEncodedPublicKey(String publicKeyString) throws CertificateException {
+        /* DER encoded Public Keys start with 0x30 (ASN.1 SEQUENCE and CONSTRUCTED), so there is
+         no leading 0x00 to drop.*/
+        PublicKey publicKey = makePublicKey(publicKeyString);
+        return new BigInteger(1, publicKey.getEncoded()).toString(16);
+    }
+
+    private PublicKey makePublicKey(final String publicKeyString) throws CertificateException {
+        try {
+            String pubKeyString = publicKeyString.replaceAll("-----.*?-----","");
+            byte[] keyBytes = Base64.decode(pubKeyString.getBytes(), Base64.DEFAULT);
+            X509EncodedKeySpec X509keySpec = new X509EncodedKeySpec(keyBytes);
+            KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+            return keyFactory.generatePublic(X509keySpec);
+        } catch (NoSuchAlgorithmException | InvalidKeySpecException e) {
+            throw new CertificateException(e);
+        }
+    }
+
+    private Boolean isValidCertificate(X509Certificate remoteCertificate)
+            throws CertificateException {
+        String remotePublicKey = getEncodedPublicKey(remoteCertificate);
```

Comment:
I have a question here. In the case that they specify both certificatesPaths and publicKeysPaths, should both _checks_ be valid or it is enough with only one?

Reply:
I think it is sufficient if one of them is valid. Same as with multiple certificates. I don't see why someone would use a separate certificate and public key to verifiy a server... 

Usually the goal is to verify that you know the server and if it can be identified with one of the provided certificates and/or public keys you know you can communicate with the server.